libminifi/test/archive-tests/CompressContentTests.cpp

```diff
@@ -91,851 +91,834 @@ class ReadCallback: public org::apache::nifi::minifi::InputStreamCallback {
-      flow1->getAttribute(FlowAttributeKey(org::apache::nifi::minifi::MIME_TYPE), mime);
-      REQUIRE(mime == "application/gzip");
-      ReadCallback callback(flow1->getSize());
-      sessionGenFlowFile.read(flow1, &callback);
-      callback.archive_read();
-      std::string flowFileName = std::string(EXPECT_COMPRESS_CONTENT);
-      std::ifstream file1;
-      file1.open(flowFileName, std::ios::in);
-      std::string contents((std::istreambuf_iterator<char>(file1)), std::istreambuf_iterator<char>());
-      std::string expectContents(reinterpret_cast<char *> (callback.archive_buffer_), callback.archive_buffer_size_);
-      REQUIRE(expectContents == contents);
-      // write the compress content for next test
-      std::ofstream file(COMPRESS_CONTENT);
-      file.write(reinterpret_cast<char *> (callback.buffer_), callback.read_size_);
-      file.close();
-      file1.close();
+    bool operator!=(const std::string& actual) const noexcept {
+      return content_ != actual;
     }
-    LogTestController::getInstance().reset();
-  } catch (...) {
+  };
+
+  std::string rawContentPath() const {
+    return raw_content_path;
+  }
+
+  std::string compressedPath() const {
+    return compressed_content_path;
+  }
+
+  RawContent getRawContent() const {
+    std::ifstream file;
+    file.open(raw_content_path, std::ios::binary);
+    std::string contents{std::istreambuf_iterator<char>(file), std::istreambuf_iterator<char>()};
+    return RawContent{std::move(contents)};
+  }
+
+  virtual ~CompressDecompressionTestController() = 0;
+};
+
+CompressDecompressionTestController::~CompressDecompressionTestController() = default;
+
+std::string CompressDecompressionTestController::tempDir;
+std::string CompressDecompressionTestController::raw_content_path;
+std::string CompressDecompressionTestController::compressed_content_path;
+
+class CompressTestController : public CompressDecompressionTestController {
+  static void initContentWithRandomData() {
+    int random_seed = 0x454;
+    std::ofstream file;
+    file.open(raw_content_path, std::ios::binary);
+
+    std::mt19937 gen(random_seed);
+    for (int i = 0; i < 100000; i++) {
+      file << std::to_string(gen() % 100);
+    }
+  }
+
+ public:
+  CompressTestController() {
+    char format[] = "/tmp/test.XXXXXX";
+    tempDir = get_global_controller().createTempDirectory(format);
+    REQUIRE(!tempDir.empty());
+    raw_content_path = utils::file::FileUtils::concat_path(tempDir, "minifi-expect-compresscontent.txt");
+    compressed_content_path = utils::file::FileUtils::concat_path(tempDir, "minifi-compresscontent");
+    initContentWithRandomData();
+  }
+
+  template<class ...Args>
+  void writeCompressed(Args&& ...args){
+    std::ofstream file(compressed_content_path, std::ios::binary);
+    file.write(std::forward<Args>(args)...);
+  }
+};
+
+class DecompressTestController : public CompressDecompressionTestController{
+ public:
+  ~DecompressTestController(){
+    tempDir = "";
```

Comment:
Could you rename the members to have an underscore suffix?
That's a fairly common practice across our source codes and I think increases readability. 

Same applies to the classes above.

Reply:
done